app/uk/gov/hmrc/breathingspaceifproxy/model/Nino.scala

```diff
@@ -16,4 +16,23 @@
 
 package uk.gov.hmrc.breathingspaceifproxy.model
 
+import cats.syntax.option._
+
 final case class Nino(value: String) extends AnyVal
+
+object Nino extends (String => Nino) {
+
+  val validNinoFormat = "^((?!(BG|GB|KN|NK|NT|TN|ZZ)|(D|F|I|Q|U|V)[A-Z]|[A-Z](D|F|I|O|Q|U|V))[A-Z]{2})[0-9]{6}[A-D ]?$"
+
+  def isValid(nino: String): Boolean = nino.matches(validNinoFormat)
+
+  def fromString(nino: String): Option[Nino] = if (isValid(nino)) Nino(nino).some else None
+
+  lazy val valid1stChars = ('A' to 'Z').filterNot(List('D', 'F', 'I', 'Q', 'U', 'V').contains).map(_.toString)
+  lazy val valid2ndChars = ('A' to 'Z').filterNot(List('D', 'F', 'I', 'O', 'Q', 'U', 'V').contains).map(_.toString)
+
+  lazy val invalidPrefixes = List("BG", "GB", "NK", "KN", "TN", "NT", "ZZ")
+
+  lazy val validPrefixes = valid1stChars.flatMap(c => valid2ndChars.map(c + _)).filterNot(invalidPrefixes.contains(_))
+  lazy val validSuffixes = ('A' to 'D').map(_.toString)
```

Comment:
Should this be placed in the test support classes (as its only used there)?

Reply:
I thought about that, but I still think this is right place as these are general logic rules which should be placed in the Nino object. At a pinch I could move them to the case class, instead of the companion object, but that makes no difference in truth.